snuba/clickhouse.py

```diff
@@ -600,20 +616,21 @@ def get_distributed_engine(cluster, database, local_table,
     }
 
 
-def get_local_table_definition():
+def get_local_table_definition(dataset):
+    table = dataset.SCHEMA.LOCAL_TABLE
     return get_table_definition(
-        settings.DEFAULT_LOCAL_TABLE, get_replicated_engine(name=settings.DEFAULT_LOCAL_TABLE)
+        table, get_replicated_engine(name=table)
     )
 
 
-def get_dist_table_definition():
-    assert settings.CLICKHOUSE_CLUSTER, "CLICKHOUSE_CLUSTER is not set."
+def get_dist_table_definition(dataset):
+    assert dataset.SCHEMA.CLICKHOUSE_CLUSTER, "CLICKHOUSE_CLUSTER is not set."
 
     return get_table_definition(
-        settings.DEFAULT_DIST_TABLE,
+        settings.dataset.SCHEMA.DIST_TABLE,
         get_distributed_engine(
-            cluster=settings.CLICKHOUSE_CLUSTER,
+            cluster=dataset.SCHEMA.CLICKHOUSE_CLUSTER,
             database='default',
```

Comment:
Actually I did not manage to find a reason why that is not a global setting. Do you have context ?

Reply:
reverting to the old way of setting the cluster. There is no strong reason to move the cluster name into the dataset for now and it will be easy to move later.